cumulusci/tasks/bulkdata/snowfakery.py

```diff
@@ -583,8 +583,10 @@ def _generate_and_load_initial_batch(self, working_directory: Path):
             self.sets_finished_while_generating_template = num_records
 
         new_template_dir = data_loader_new_directory_name(template_dir, self.run_until)
-        shutil.move(template_dir, new_template_dir)
-        template_dir = new_template_dir
+        # rename only if new_template_dir does not match template_dir
+        if Path(template_dir).resolve() != Path(new_template_dir).resolve():
```

Comment:
Both `template_dir` and `new_template_dir` are already `Path` instances, so we can simplify to:
```suggestion
        if template_dir.resolve() != new_template_dir.resolve():
```